cmd/fs-v1-append-parts.go

```diff
@@ -0,0 +1,170 @@
+package cmd
+
+import (
+	"errors"
+	"io"
+	"reflect"
+	"sync"
+	"time"
+)
+
+// Error sent by appendRoutine when there are holes in parts.
+// For ex. let's say client uploads part-2 before part-1 in which case we
+// can not append and have to wait till part-1 is uploaded. Hence we return
+// this error. Currently this error is not used in the caller.
+var errPartsMissing = errors.New("required parts missing")
+
+// Error sent when appendRoutine has waited long enough and timedout.
+var errAppendRoutineTimeout = errors.New("timeout")
+
+// Timeout value for the appendRoutine go-routine.
+var appendRoutineTimeout = 24 * 60 * 60 * time.Second
+
+// Holds a map of uploadID->appendRoutine
+type appendPartFS struct {
+	appendRoutineMap map[string]appendRoutineInfo
+	mu               sync.Mutex
+}
+
+// Input to the appendRoutine
+type appendRoutineInput struct {
+	meta  fsMetaV1   // list of parts that need to be appended
+	errCh chan error // error sent by appendRoutine
+}
+
+// Identifies an appendRoutine.
+type appendRoutineInfo struct {
+	inputCh   chan appendRoutineInput
+	timeoutCh chan struct{} // closed by appendRoutine when it timesout
+	endCh     chan struct{} // closed after complete/abort of upload to end the appendRoutine
+}
+
+// Called after a part is uploaded so that it can be appended in the background.
+func (m *appendPartFS) processPart(disk StorageAPI, bucket, object, uploadID string, meta fsMetaV1) chan error {
```

Comment:
we can say what it is going to do 'processPart' doesn't convey what it is going to do. It is too generic.